### PR TITLE
Issue #13853: migrate FinalLocalVariable xdoc to use property macro

### PR DESCRIPTION
**Pull request by AayushSaini101**
Related:  https://github.com/checkstyle/checkstyle/issues/13853

Old PR: https://github.com/checkstyle/checkstyle/pull/13878

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java [hunk 1]
```diff
@@ -68,7 +69,10 @@ public class PropertiesMacro extends AbstractMacro {
     /** A newline with 20 spaces of indentation. */
     private static final String INDENT_LEVEL_20 = SiteUtil.getNewlineAndIndentSpaces(20);
 
-    /** The name of the current module being processed. */
+    /** The property being process with all tokens. */
+    private static final String TOKENS_PROPERTY = SiteUtil.TOKENS;
+
+    /** The file of the current module being processed. */
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:75]: Why is this changed?

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java [hunk 2]
```diff
@@ -165,13 +169,34 @@ private static void writeTablePropertiesRows(Sink sink)
         final Map<String, DetailNode> propertiesJavadocs = SiteUtil
                 .getPropertiesJavadocs(properties, currentModuleName, currentModuleFile);
 
-        for (String property : properties) {
+        final List<String> orderProperties = orderProperties(properties);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:172]:
```suggestion
        final List<String> orderedProperties = orderProperties(properties);
```

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java [hunk 3]
```diff
@@ -165,13 +169,34 @@ private static void writeTablePropertiesRows(Sink sink)
...
+    private static List<String> orderProperties(Set<String> properties) {
+
+        final List<String> validProperties = new LinkedList<>(properties);
+
+        if (validProperties.remove(TOKENS_PROPERTY)) {
+            validProperties.add(TOKENS_PROPERTY);
+        }
+
+        return validProperties;
+
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:197]: Please return immutable list here, as suggested in other PR:
```suggestion
      if (properties.remove(TOKENS_PROPERTY)) {
            properties.add(TOKENS_PROPERTY);
        }
        return List.copyOf(properties);
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java (hunk 1 above)
Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:75]: @nrmancuso This property is used to correct the ordered properties.

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java (hunk 2 above)
Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java]: Done @nrmancuso

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java (hunk 3 above)
Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java]: Rewrite the conditions, direclty removing from the set and adding the set causing an error, hence creating list first and then after performing the operation returning immutable list

File: src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:75]: It might looks now different than it was before.
But now I see no difference, and GitHub shows diff with extra lines change but content is not changed

Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:78]: Added @romani

Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:78]: Why did we change `name` to `file`? I would expect something that is called a `file` to be of type `File`. Please revert this change.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:78]: @AayushSaini101, please restore this line as it was before.

Comment (AayushSaini101) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java]: @nrmancuso @romani Done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
